File: packages/jexlang-java/src/main/java/com/jexlang/java/eval/errors/UndefinedFunctionError.java
package com.jexlang.java.eval.errors;

public class UndefinedFunctionError extends JexLangRuntimeError {
    private final String functionName;

    public UndefinedFunctionError(String functionName) {
        super("Undefined function: " + functionName);
        this.functionName = functionName;
    }

    public String getFunctionName() {
        return functionName;
    }
}

File: packages/jexlang-swift/jexlang-swift/eval/errors/UndefinedFunctionError.swift
//
//  UndefinedFunctionError.swift
//  jexlang-swift
//
//  Created by Vinay Kumar on 03/10/25.
//

import Foundation

public class UndefinedFunctionError: JexLangRuntimeError {
    public let functionName: String
    
    init(functionName: String) {
        self.functionName = functionName
        super.init(message: "Undefined function: " + functionName)
    }
    
    public func getFunctionName() -> String {
        return self.functionName;
    }
}
